src/tiled/tilecollisiondock.cpp

```diff
@@ -127,6 +136,51 @@ TileCollisionDock::~TileCollisionDock()
     setTile(nullptr);
 }
 
+/**
+ * @brief TileCollisionDock::autoDetectMask
+ *
+ * Automatically detect the extents of the tile and append a simple rectanglular collision mask.
+ */
+void TileCollisionDock::autoDetectMask()
+{
+    // Iterate over the pixels, looking for empty rows
+    // sourced from: https://stackoverflow.com/a/3722160/2431627
+    QImage image = mTile->image().toImage();
+    int left = image.width(), right = 0, top = image.height(), bottom = 0;
+    for (int y = 0; y < image.height(); ++y) {
+        QRgb *row = (QRgb*)image.scanLine(y);
+        bool rowFilled = false;
+        for (int x = 0; x < image.width(); ++x) {
+            if (qAlpha(row[x])) {
+                rowFilled = true;
+                right = std::max(right, x);
+                if (left > x) {
+                    left = x;
+                    x = right; // shortcut to only search for new right bound from here
+                }
+            }
+        }
+        if (rowFilled) {
+            top = std::min(top, y);
+            bottom = y;
+        }
+    }
+
+    // Create a group for collision objects if none exists
+    if (!mTile->objectGroup()) {
+        mTile->setObjectGroup(new ObjectGroup);
```

Comment:
You can't just directly modify the tile, since that way the operation cannot be undone.

For the undo system to work in the tile collision editor, the undo stack of the `mDummyMapDocument` should be used. This dummy map is guaranteed to already have an object layer (see `TileCollisionDock::setTile`). You'd only need to use something like the following to add the object:
```c++
    auto objectGroup = static_cast<ObjectGroup*>(mDummyMapDocument->map()->layerAt(1));
    auto command = new AddMapObjects(mDummyMapDocument, objectGroup, newObject);
    mDummyMapDocument->undoStack()->push(command);
```
That way the UI will also update automatically, since that is triggered by the undo command.

(Note that you'll need to rebase your changes, since `AddMapObjects` used to be called `AddMapObject` and could only add one object, but I've recently changed that.)